### PR TITLE
ThinkThoroughly powered by Prophet

### PR DESCRIPTION
**Pull request by kongzii**

### CODE REVIEW

**Review by evangriffiths — APPROVED** [review]

**Comment by kongzii** [comment]
Runned this on the question Martin asked last time about:

```
Will the fugitive 'Cryptoqueen' Ruja Ignatova be captured by July 17, 2024?
    TT:                  p_yes=0.15      confidence=0.85
    TT Prophet GPT4:     p_yes=0.35      confidence=0.7
    TT Prophet GPT3.5:   p_yes=0.1       confidence=0.8
Will the fugitive 'Cryptoqueen' Ruja Ignatova be captured by July 20, 2024?
    TT:                  p_yes=0.1       confidence=0.85
    TT Prophet GPT4:     p_yes=0.35      confidence=0.75
    TT Prophet GPT3.5:   p_yes=0.2       confidence=0.75
Will the fugitive 'Cryptoqueen' Ruja Ignatova be captured by July 20, 2025?
    TT:                  p_yes=0.45      confidence=0.7 
    TT Prophet GPT4:     p_yes=0.25      confidence=0.65
    TT Prophet GPT3.5:   p_yes=0.1       confidence=0.8
Will the fugitive 'Cryptoqueen' Ruja Ignatova be captured by July 20, 2050?
    TT:                  p_yes=0.55      confidence=0.7 
    TT Prophet GPT4:     p_yes=0.75      confidence=0.85
    TT Prophet GPT3.5:   p_yes=0.3       confidence=0.8 
```

Seems that original TT is doing better (not perfect) job on keeping probabilities in line within time frames. But Prophet should have better accuracy in the end... So let's see.

File: prediction_market_agent/agents/think_thoroughly_agent/deploy.py [hunk 1]
```diff
@@ -24,6 +27,16 @@ def before(self, market_type: MarketType) -> None:
...
 
 
+class DeployableThinkThoroughlyAgent(DeployableThinkThoroughlyAgentBase):
+    agent_class = ThinkThoroughlyWithItsOwnResearch
+    model: str = "gpt-4-turbo-2024-04-09"
+
+
+class DeployableThinkThoroughlyProphetResearchAgent(DeployableThinkThoroughlyAgentBase):
+    agent_class = ThinkThoroughlyWithPredictionProhpetResearch
+    model: str = "gpt-3.5-turbo-0125"
```
Comment (kongzii) [inline prediction_market_agent/agents/think_thoroughly_agent/deploy.py:37]: Prophet itself seems to work better with 3.5, so why not use it here as well.

File: scripts/agent_app.py [hunk 2]
```diff
@@ -26,102 +25,118 @@
...
+    DeployableThinkThoroughlyProphetResearchAgent,
+]
 
 add_sink_to_logger()
 
-st.title("Agent's decision-making process")
 
-with st.sidebar:
-    streamlit_login()
+def agent_app() -> None:
```
Comment (kongzii) [inline scripts/agent_app.py:48]: Multiprocessing errors out, if the streamlit application is not inside of a function and behind a `if __name__ == "__main__"` block.

File: prediction_market_agent/agents/think_thoroughly_agent/deploy.py [hunk 3]
```diff
@@ -24,6 +27,16 @@
...
```
[the remark this answers is not in the document]
Comment (evangriffiths) [reply prediction_market_agent/agents/think_thoroughly_agent/deploy.py:37]: coderabbit has good eyesight!

**Pull request merged**